### PR TITLE
Soft distillation training support

### PR DESCRIPTION
**Pull request by gagika**
# Description

This PR introduces `train_distill.py`, a new training script that implements a **soft distillation workflow** by integrating the [Tunix](https://github.com/google/tunix) library with MaxText.

**Context & Problem Solved:**
This script enables a soft distillation process where a smaller, trainable **Student** model mimics the output distribution of a larger, frozen **Teacher** model on general data.

**Key Implementation Details:**
* **Dual Model Loading:** Initializes two distinct MaxText models (Student and Teacher) in the same process using `pyconfig` with `student_overrides` and `teacher_overrides` to support different architectures.
* **Tunix Integration:** Wraps MaxText models in `TunixMaxTextAdapter` to expose a standard interface compatible with the Tunix `DistillationTrainer`.

## Run distillation

```bash
python3 -m src.MaxText.distillation.train_distill src/MaxText/configs/distillation.yml   run_name=distillation_test   base_output_directory=${BASE_OUTPUT_DIRECTORY}   checkpoint_period=2000   hf_access_token=$HF_TOKEN   log_period=100  save_checkpoint_on_completion=True 
```

# Tests

I have tested this script end-to-end on a TPU VM.
Verified that Student and Teacher models load with distinct configurations   

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by richjames0 — APPROVED** [review]
lgtm

File: src/MaxText/configs/distillation.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+distill_alpha: 0.5
+distill_temperature: 1.0
+
+# --- Dataset & Tokenizer ---
+hf_path: "OptimalScale/ClimbMix"
+dataset_type: "hf"
+tokenizer_path: "meta-llama/Llama-3.1-8B"
+tokenizer_type: "huggingface"
+
+# dataset_path: "gs://max-datasets-rogue"
```
Comment (NicoGrande) [inline src/MaxText/configs/distillation.yml]: Can we remove these commented lines?

File: src/MaxText/distillation/train_distill.py [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+  if student_config.vocab_size != teacher_config.vocab_size:
+    raise ValueError(
+        f"Vocab size mismatch! Student: {student_config.vocab_size}, Teacher: {teacher_config.vocab_size}. "
+        "Distillation requires matching vocabularies."
+    )
+
+  # 1. Setup Mesh
+  devices = jax.devices()
+  devices_array = maxtext_utils.create_device_mesh(student_config, devices)
+  mesh = jax.sharding.Mesh(devices_array, student_config.mesh_axes)
```
Comment (NicoGrande) [inline src/MaxText/distillation/train_distill.py:382]: Is my understanding correct that both models will be collocated on the same Mesh? Will this need to be extended to disaggregated to work for different size student / teacher models for example?

**Review by NicoGrande — APPROVED** [review]
LGTM!

File: src/MaxText/configs/distillation.yml (hunk 1 above)
Comment (gagika) [reply src/MaxText/configs/distillation.yml]: done

File: src/MaxText/distillation/train_distill.py (hunk 2 above)
Comment (gagika) [reply src/MaxText/distillation/train_distill.py:382]: yes, it's collocated setup on the same mesh, we can still run different model sizes on the same mesh with possibly different sharding. 

disaggregated setup not yet planned / supported, we can extend in the future if needed.

**Review by sheng-li — APPROVED** [review]
LGTM! 
btw, updated the PR description slightly. PTAL
